Basic-Captions/js/app.js

```diff
@@ -113,6 +115,36 @@ async function postData(url='', data={}){
   }
 }
 
+const selfSubscribe = (session) => {
+  const captionSub = session.subscribe(publisher.stream, document.createElement('div'),
+   {testNetwork: true})
+   // Note that testNetwork will be fixed in 2.25.1
+  captionSub.setAudioVolume(0)
+
+  captionSub.on('captionReceived', (event) => {
+    console.log(event.caption)
+    const captionText = event.caption;
+      const publisherContainer = OT.publishers.find().element;
+      const [publisherWidget] = publisherContainer.getElementsByClassName('OT_widget-container');
+    
+      const oldCaptionBox = publisherWidget.querySelector('.caption-box');
+      if (oldCaptionBox) oldCaptionBox.remove();
+    
+      const captionBox = document.createElement('div');
+      captionBox.classList.add('caption-box');
+      captionBox.textContent = captionText;
+    
+      // remove the captions after 5 seconds
+      const removalTimerDuration = 5 * 1000;
+      clearTimeout(captionsRemovalTimer);
+      captionsRemovalTimer = setTimeout(() => {
+        captionBox.textContent = '';
+      }, removalTimerDuration);
```

Comment:
nit: this part is the same in `subscriber.on('captionReceived')` and the new `captionSub.on('captionReceived').` do you think it would be possible to move this block to a separate function and call it inside of both?

Reply:
Good call

Reply:
For legibility, may need to put a black background behind the text.